cmd/soroban-cli/src/commands/config/locator.rs

```diff
@@ -280,15 +280,26 @@ impl Args {
         Ok(alias_data.ids.get(network_passphrase).cloned())
     }
 
-    pub fn resolve_contract_id(
+    pub fn resolve_contract_id_as_string(
         &self,
         alias_or_contract_id: &str,
         network_passphrase: &str,
-    ) -> Result<[u8; 32], Error> {
+    ) -> Result<String, Error> {
         let contract_id = self
             .get_contract_id(alias_or_contract_id, network_passphrase)?
             .unwrap_or_else(|| alias_or_contract_id.to_string());
 
+        Ok(contract_id)
+    }
+
+    pub fn resolve_contract_id(
+        &self,
+        alias_or_contract_id: &str,
+        network_passphrase: &str,
+    ) -> Result<[u8; 32], Error> {
+        let contract_id =
+            self.resolve_contract_id_as_string(alias_or_contract_id, network_passphrase)?;
+
```

Comment:
We could collapse these two functions into one that returns a `StrKey::Contract` which can then be converted into a string or byte array.

Reply:
I wanted to avoid too many changes on that module, as I don't fully grasp it. 

Can `StrKey::Contract` be resolved from just the C-address? The `::from_string` function lacks documentation, but it seems like it waits for a contract payload, right?

* https://github.com/stellar/rs-stellar-strkey/blob/v0.0.8/src/strkey.rs#L204

Reply:
You can either use the from_string method or construct it from the `[u8; 32]`. So it's a small change to `resolve_contract_id`.


```rust
#[derive(Clone, Copy, Hash, PartialEq, Eq, PartialOrd, Ord)]
pub struct Contract(pub [u8; 32]);
```

Since the inner tuple field is public you can construct it like `stellar_strkey::Contract(<arr>)`. Then from the caller.
```rust
let contract_strkey = resolve_contract_id(...)?;
let contract_id = contract_strkey.0;
let contract_id_str = contract_strkey.to_string();
```

Reply:
ah, that makes sense. Let me try it! :)